pkg/vsphere/vm/vm.go

```diff
@@ -198,30 +199,30 @@ func (vm *VirtualMachine) WaitForKeyInExtraConfig(ctx context.Context, key strin
 					if key == value.GetOptionValue().Key {
 						detail = value.GetOptionValue().Value.(string)
 						if detail != "" && detail != "<nil>" {
-							return true
+							exitEarly = true
 						}
 						break // continue the outer loop as we may have a powerState change too
 					}
 				}
 			case types.VirtualMachinePowerState:
 				if v != types.VirtualMachinePowerStatePoweredOn {
-					// Give up if the vm has powered off
 					msg := "powered off"
 					if v == types.VirtualMachinePowerStateSuspended {
 						// Unlikely, but possible if the VM was suspended out-of-band
 						msg = string(v)
 					}
 					poweredOff = fmt.Errorf("container VM has unexpectedly %s", msg)
-					return true
+					exitEarly = true
 				}
 			}
-
 		}
-		return false
+		return exitEarly
 	}
 
+	// if poweredOff != nil && detail == "" then assume tether did not initialize.
+	// if poweredOff != nil && detail != "" then assume tether did initialize but VM shutdown quickly for valid reasons
 	err := vm.WaitForExtraConfig(ctx, waitFunc)
-	if err == nil && poweredOff != nil {
+	if err == nil && poweredOff != nil && detail == "" {
```

Comment:
I think the check `detail == ""` can help to avoid powerdOff error if the value is already set by tether.
But what's the benefit to set exitEarly above? If you want, maybe you could just return at line 202, https://github.com/vmware/vic/pull/5938/files#diff-818b18702ad1cb5203eeaccc54071e09R202